jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -492,40 +492,37 @@ ecma_builtin_array_prototype_object_push (const ecma_value_t *argument_list_p, /
                                           ecma_object_t *obj_p, /**< array object */
                                           uint32_t length) /**< array object's length */
 {
-  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
-
   ecma_number_t n = (ecma_number_t) length;
 
   /* 5. */
-  for (uint32_t index = 0;
-       index < arguments_number && ecma_is_value_empty (ret_value);
-       index++, n++)
+  for (uint32_t index = 0; index < arguments_number; index++, n++)
   {
-    /* 5.a */
-    ecma_value_t e_value = argument_list_p[index];
-
     /* 5.b */
     ecma_string_t *n_str_p = ecma_new_ecma_string_from_number (n);
 
-    ECMA_TRY_CATCH (put_value, ecma_op_object_put (obj_p, n_str_p, e_value, true), ret_value);
-    ECMA_FINALIZE (put_value);
+    ecma_value_t put_value = ecma_op_object_put (obj_p, n_str_p, argument_list_p[index], true);
```

Comment:
Can't we use `ecma_op_object_put_by_index` here?

Reply:
`ecma_op_object_put_by_index` is designed to handle only uint32_t indices, but at this point `ecma_number_t n = (ecma_number_t) length;` can overflow in the UINT32_MAX limit.